### PR TITLE
Recover and add new tests to TicTacToeEngine

### PR DESCRIPTION
**Pull request by wujuu**

### CODE REVIEW

**Comment by wujuu** [comment]
Closes #23

File: tests/environments/test_board.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from unittest import TestCase
+from environments.tic_tac_toe.tic_tac_toe_engine import _Board
+
+
+class TestBoard(TestCase):
+    def setUp(self):
+        size = 5
+        marks = [0, 1, 2]
+        marks_required = 3
```
Comment (JanKapala) [inline tests/environments/test_board.py]: What about testing different sizes and marks_required?

File: tests/environments/test_tic_tac_toe_engine.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+from environments.tic_tac_toe.tic_tac_toe_engine import TicTacToeEngine
+from environments.tic_tac_toe.tic_tac_toe_engine_utils import Player, Winning, IllegalMoveError, TicTacToeAction, \
+    TicTacToeActionSpace
+
+
+class TestTicTacToeEngine(TestCase):
+    def setUp(self):
+        self.no_of_players = 3
+        self.board_size = 5
+        self.marks_required = 3
```
Comment (JanKapala) [inline tests/environments/test_tic_tac_toe_engine.py]: Same as above

File: tests/environments/test_tic_tac_toe_engine_additional.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+from environments.tic_tac_toe.tic_tac_toe_engine_utils import Player, Winning, IllegalMoveError, TicTacToeAction, \
+    TicTacToeActionSpace
+from tests.environments.test_tic_tac_toe_engine import TestTicTacToeEngine
+
+
+class TestTicTacToeEngineAdditional(TestTicTacToeEngine):
+    def setUp(self):
+        self.no_of_players = 2
+        self.board_size = 3
+        self.marks_required = 3
```
Comment (JanKapala) [inline tests/environments/test_tic_tac_toe_engine_additional.py]: Same as above

File: tests/environments/test_tic_tac_toe_engine.py (hunk 2 above)
Comment (wujuu) [reply tests/environments/test_tic_tac_toe_engine.py]: It is tested in testTicTacToeAdditional.py

File: tests/environments/test_board.py (hunk 1 above)
Comment (wujuu) [reply tests/environments/test_board.py]: Same as below

File: tests/environments/test_tic_tac_toe_engine.py (hunk 2 above)
Comment (JanKapala) [reply tests/environments/test_tic_tac_toe_engine.py]: So you're saying that you have less test cases than dimensions in space (players,size,marks)... It's not extensive testing.

Comment (wujuu) [reply tests/environments/test_tic_tac_toe_engine.py]: So we should hire some students to test our code extensively

**Pull request merged**